addons/website_sale_stock/static/src/xml/website_sale_stock_product_availability.xml

```diff
@@ -1,29 +1,25 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
 <templates>
-
     <t t-name="website_sale_stock.product_availability">
-        <t t-if="product_type == 'product' and _.contains(['always', 'threshold'], inventory_availability)">
-            <t t-if="virtual_available gt 0">
-                <div t-if="inventory_availability == 'always'" t-attf-class="availability_message_#{product_template} text-success mt16">
-                    <t t-esc="virtual_available_formatted" /> <t t-esc="uom_name" /> available
-                </div>
-                <t t-if="inventory_availability == 'threshold'">
-                    <div t-if="virtual_available lte available_threshold" t-attf-class="availability_message_#{product_template} text-warning mt16">
-                        <i class="fa fa-exclamation-triangle" title="Warning" role="img" aria-label="Warning"/>
-                        <t t-esc="virtual_available_formatted" /> <t t-esc="uom_name" /> available
-                    </div>
-                    <div t-if="virtual_available gt available_threshold" t-attf-class="availability_message_#{product_template} text-success mt16">In stock</div>
-                </t>
-            </t>
+        <t t-set="display_informations" t-value="product_type == 'product' and is_published"/>
+        <t id="stock_informations" t-if="display_informations">
```

Comment:
`id` on a `t` is a bit weird, as it is only for xpath purpose, can you get rid of it and just xpath `#available_message//after` instead?
Result should be the same, as the conditions are differents, both can't be displayed anyway.